src/m365/commands/login.ts

```diff
@@ -243,7 +243,13 @@ class LoginCommand extends Command {
       }
     }
     finally {
-      if (!args.options.ensure || (args.options.ensure && !auth.service.connected)) {
+      if (!args.options.ensure ||
+        (args.options.ensure && !auth.service.connected) ||
+        (args.options.ensure && (args.options.userName && args.options.userName !== auth.service.userName)) ||
+        (args.options.ensure && (args.options.certificateFile && (auth.service.certificate !== fs.readFileSync(args.options.certificateFile as string, 'base64')))) ||
+        (args.options.ensure && (args.options.appId && args.options.tenant && (args.options.appId !== auth.service.appId) && (args.options.tenant !== auth.service.tenant))) ||
+        (args.options.ensure && args.options.authType !== 'identity' && auth.service.authType !== AuthType.DeviceCode)
```

Comment:
Why are we checking for these auth types specifically?

Reply:
Let's check if we're talking about the same:

When I said: _Don't start login flow if the CLI is signed in using Managed Identity_ I mean: _check if the CLI is currently signed in using a Managed Identity. If it is, all good. If it's signed in using a different method or not signed in at all, start the auth flow_. The same applies to Device Code. I didn't mean to exclude both auth flows altogether.

Agreed?

Reply:
Ok, I'll remove the check for `DeviceCode` and `identity`.

Regarding `identity`, there is a check if another username is used and regarding `DeviceCode`, i guess the check for `auth.service.connected` should be enough

Reply:
> and regarding `DeviceCode`, i guess the check for `auth.service.connected` should be enough

Not quite. If there's no auth method specified, then the logic should verify that the default (meaning DeviceCode) auth type has been used previously.